### PR TITLE
Commented lines for DataParallel and explicitly refer to cuda:0 as de…

### LINKED ISSUE #1: Run with multiple GPU
Verify use of `torch.nn.DataParallel`

### PR DESCRIPTION
**Pull request by allaffa**
Commented lines for DataParallel and explicitly refer to cuda:0 as device to use

Closes #1

### CODE REVIEW

File: utilities/utils.py
Comment (streeve) [inline utilities/utils.py:134]: Should we add another condition where we do want to use multiple GPU or just wait until we use distributed parallelism?

File: utilities/utils.py
Comment (streeve) [inline utilities/utils.py:103]: Did the code work before if using the number of GPU available?

**Review by streeve — COMMENTED** [review]
Great that this works! Just a couple questions

File: utilities/utils.py
Comment (allaffa) [reply utilities/utils.py:134]: There are two types of data parallelism. 

DistributedDataParallelism (newer) -> multi-processing
DataParallelism (older) -> multi-threading

I would say that DistributedDataParallelism is more compelling to take advantage of distributed resources to perform the biggest chunk of the parallelization. 
DataParallelism could be combined with DistributedDataParallelism, but I would not make that our priority.

File: utilities/utils.py
Comment (allaffa) [reply utilities/utils.py:103]: device = "cuda" also works if the number of visible CUDA devices is bigger than 1. However, it is not clear to me whether one CUDA devices is randomly picked among the ones available, or if by default "cuda:0" is the one always picked. 

"cuda:0" is the local number of GPUs, meaning that if
export CUDA_VISIBLE = 2,3,4
GPU with index = 2 will be "cuda:0"

File: utilities/utils.py
Comment (streeve) [reply utilities/utils.py:134]: To clarify my own understanding: DDP is distributed memory ("MPI" on one or more nodes), DP is only capable of single node, but still "MPI" parallel

File: utilities/utils.py
Comment (streeve) [reply utilities/utils.py:103]: It's probably not crucial since the plan is to entirely switch to DDP, but should we keep the commented DP or just remove?

Comment (allaffa) [reply utilities/utils.py:103]: I would keep it just as a placeholder for now.

**Comment by allaffa** [comment]
@streeve 

Sorry for taking so long. I rebased the `fix_device_bug` branch on the master, since they diverged after merging your pull request. You can proceed and complete the pull request if you agree with what I propose.

**Comment by streeve** [comment]
> Sorry for taking so long. I rebased the `fix_device_bug` branch on the master, since they diverged after merging your pull request. You can proceed and complete the pull request if you agree with what I propose.

No worries - I re-rebased and will merge after it passes

**Review by streeve — APPROVED** [review]

**Pull request merged**
